Advection problem

AdelekeBankole opened the pull request.

AdelekeBankole committed:
Move conformal maps to library

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update src/Basis/Constructors.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update src/Basis/Constructors.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
Update examples/ex010_advection.jl

Co-authored-by: Jed Brown <jed@jedbrown.org>

AdelekeBankole committed:
delete ex011_advection_mapped.jl

AdelekeBankole committed:
Merge branch 'transplanted_quadrature' of https://github.com/AdelekeBankole/LFAToolkit.jl into transplanted_quadrature

AdelekeBankole committed:
Fix g_prime syntax

AdelekeBankole committed:
Merge branch 'transplanted_quadrature' of https://github.com/AdelekeBankole/LFAToolkit.jl into transplanted_quadrature

AdelekeBankole committed:
advection problem

AdelekeBankole committed:
remove dead code and fix formatting

AdelekeBankole committed:
remove dead code

AdelekeBankole committed:
fix styles

AdelekeBankole committed:
fix styles

AdelekeBankole committed:
add transform to library

AdelekeBankole committed:
verify doctests

AdelekeBankole committed:
doc for advection operator

AdelekeBankole committed:
add advection.md in examples

AdelekeBankole committed:
update references

AdelekeBankole committed:
fix advectionoperator in gallery

AdelekeBankole committed:
some failing tests in transforms

AdelekeBankole commented:
I still have some failing tests in src/Basis/Constructors.jl 
Could you help me check in the CI?

jedbrown commented:
Start with the first error.
```
 │ ERROR: UndefVarError: mapping not defined
```
https://github.com/jeremylt/LFAToolkit.jl/runs/7102379147?check_suite_focus=true#step:5:122
The variable `mapping` is not defined in the code snippet printed above. https://github.com/jeremylt/LFAToolkit.jl/runs/7102379147?check_suite_focus=true#step:5:93

Many of the subsequent errors are similar. Just take them one at a time, find the place in the code it's telling you about, and fix the error (defining the variable that's undefined where it's used above).

I think the issue here is that it needs to be a keyword argument. https://github.com/jeremylt/LFAToolkit.jl/runs/7102379147?check_suite_focus=true#step:5:322
If you know a function has been called successfully elsewhere, you can use `rg` or `git grep` to find other occurrences. Note that Control-Shift-F in vscode searches the project.

In other cases, you get this, which must be a bug in the code, seemingly introduced because we've run all this before in an earlier version. You could check old history (with or without `git bisect`) to find how/when this bug was introduced. https://github.com/jeremylt/LFAToolkit.jl/runs/7102379147?check_suite_focus=true#step:5:590

AdelekeBankole committed:
add mapping

AdelekeBankole committed:
still some test passing problems

AdelekeBankole committed:
working on test passing issues

AdelekeBankole commented:
The main problem is a domain error related to the haletrefethenstriptransformation

AdelekeBankole committed:
some errors with mapped weights

jedbrown committed:
isapprox in tests when comparing to small values

jedbrown committed:
transformquadrature: fix arg documentation and doctest

The Hale-Trefethen and Kosloff-Tal-Ezer transformations do not
accurately integrate constants except in the limit. The sausage(9) map
is pretty good once you have at least 5 Gauss points.

jedbrown commented on src/Basis/Constructors.jl:
```diff
@@ -124,9 +124,9 @@ Transformed quadrature by applying a smooth mapping = (g, gprime) from the origi
-mpoints, mweights = LFAToolkit.transformquadrature(points, true, mapping);
+points, weights = LFAToolkit.gaussquadrature(5)
+mapping = sausagetransformation(9)
+mpoints, mweights = transformquadrature(points, weights, mapping)
```
The domain error was because `weights` was incorrectly documented (and used here) as boolean, when in fact it needs to be an array of weights.

I switched to sausage here because it more accurately integrates the constant on coarse grids (few quadrature points); the others need many points to make that true.

jedbrown commented on test/runtests.jl:
```diff
@@ -36,25 +36,14 @@ end
-
-    #    @test min(eigenvalues...) ≈ 0.008379422444571976
-    #    @test max(eigenvalues...) ≈ 0.17361111111111088
+    @test min(eigenvalues...) ≈ -30.84740977051096
+    @test max(eigenvalues...) ≈ 30.847409770510946
```
It looks like these reference values were computed for a different configuration.

AdelekeBankole committed:
Fix min and max eigenvalues

AdelekeBankole committed:
Fix accuracy test

AdelekeBankole committed:
Merge branch 'transplanted_quadrature' of https://github.com/AdelekeBankole/LFAToolkit.jl into transplanted_quadrature

AdelekeBankole committed:
Update docs/src/examples/advection.md

Co-authored-by: Jed Brown <jed@jedbrown.org>

jedbrown commented:
This PR sources from the same branch as #48, but targets a `transplanted_quadrature` branch in Jeremy's repo instead of `main`. I'll close this so we stop getting double-notifications about the same work. Let's use #48 to continue the review.

AdelekeBankole commented:
Yes I agree we close it and use #48 henceforth
